lib/ohai/plugins/hostname.rb

```diff
@@ -138,9 +138,23 @@ def sigar_is_available?
   end
 
   collect_data(:solaris2) do
+    require 'socket'
+
     machinename from_cmd("hostname")
     hostname from_cmd("hostname")
-    fqdn resolve_fqdn
+    fqdn_lookup = Socket.getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first[2]
+    if fqdn_lookup.split('.').length > 1
+      # we recieved an fqdn
+      fqdn fqdn_lookup
+    else
+      # default to assembling one
```

Comment:
We should probably use `hostname` to start with and if it has dots in it, just use that.

On Solaris `hostname` will not have dots, then as a third step we can append the `domainname`.